plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/WarPublisher.java

```diff
@@ -58,14 +60,19 @@ public static void publishExploded(IProject project, IPath destination,
     }
 
     SubMonitor subMonitor = SubMonitor.convert(monitor, 100);
-    subMonitor.setTaskName(Messages.getString("task.name.publish.war"));
+    subMonitor.setTaskName(Messages.getString("task.name.publish.war")); //$NON-NLS-1$
 
     IModuleResource[] resources =
         flattenResources(project, safeWorkDirectory, subMonitor.newChild(10));
-    PublishUtil.publishFull(resources, destination, subMonitor.newChild(90));
+    if (resources.length == 0) {
+      IStatus error = StatusUtil.error(WarPublisher.class, project.getName()
+          + " has no resource to publish"); //$NON-NLS-1$
```

Comment:
I think "resource" would be plural here?

Reply:
Thanks. I've used this as an opportunity to educate myself on this topic. I think I've conclusively understood the usage.